src/_pytest/python.py

```diff
@@ -681,22 +682,24 @@ def __init__(
             session=session,
             nodeid=nodeid,
         )
-        self.name = self.path.parent.name
+        self.name = self.path.name
```

Comment:
The `path` is now the directory, not the `__init__.py`. The name is the name of the directory, so no need for the `parent` anymore.